File: src/main/java/github/macro/pieces/Thief.java
package github.macro.pieces;

import github.macro.Direction;
import github.macro.console.Colour;
import org.jetbrains.annotations.NotNull;

import static github.macro.Direction.*;

/**
 * Created by Macro303 on 2018-02-09.
 */
public class Thief extends Piece {
	@NotNull
//	private static final String symbol = "\uD83D\uDDE1";
	private static final String symbol = "T";
	private static final int maxDistance = 1;
	@NotNull
	private static final Direction[] validDirections = new Direction[]{NORTH, NORTH_EAST, EAST, SOUTH_EAST, SOUTH, SOUTH_WEST, WEST, NORTH_WEST};

	public Thief(@NotNull Colour colour) {
		super(colour, maxDistance, symbol, validDirections);
	}

	public Thief() {
		super(maxDistance, symbol, validDirections);
	}

	@Override
	public String toString() {
		return "Thief{} " + super.toString();
	}
}

File: src/main/kotlin/github/macro/pieces/Thief.kt
package github.macro.pieces

import github.macro.Direction
import github.macro.players.Player

/**
 * Created by Macro303 on 2018-02-09.
 */
class Thief(player: Player) : Piece(
	player = player,
	distance = 1,
	symbol = "\uD83D\uDDE1",
	directions = arrayOf(
		Direction.NORTH,
		Direction.NORTH_EAST,
		Direction.EAST,
		Direction.SOUTH_EAST,
		Direction.SOUTH,
		Direction.SOUTH_WEST,
		Direction.WEST,
		Direction.NORTH_WEST
	)
) {
	//	private static final String symbol = "\uD83D\uDDE1";
	//	private static final String symbol = "T";
	override fun toString(): String = "Thief() ${super.toString()}"
}